src/modm/architecture/driver/atomic/queue_impl.hpp

```diff
@@ -122,4 +122,12 @@ modm::atomic::Queue<T, N>::pop()
 	this->tail = tmptail;
 }
 
+template<typename T, std::size_t N>
+void
+modm::atomic::Queue<T, N>::clear()
+{
+	this->head = 0;
+	this->tail = 0;
```

Comment:
This is not interrupt safe though?

Reply:
Uhmm, would a `modm::atomic::Lock l;` improve the situation?

Reply:
Yes, but only if clear is called from outside an interrupt. If clear is called inside an interrupt, there's no guarantee that the code outside the interrupt won't write these members back with their `tmp` values.
The atomic queue is not truly atomic, it's merely a single-producer, single-consumer queue that works one sided with non-nested interrupts. It's really very AVR specific, and should be rewritten completely for Cortex-M (or better, imported from an existing project).

Reply:
The single-producer, single-consumer queue in boost lockfree also has [a non-thread safe `reset()`](https://github.com/boostorg/lockfree/blob/af906f5745e1d6c024db0b4919b56de7f2f9b70b/include/boost/lockfree/spsc_queue.hpp#L350). It is basically equivalent to the proposed `clear()` method. Not sure if it is worth adding it, though.